File: src/main/java/net/thenamedev/legendarena/commands/Particle.java
package net.thenamedev.legendarena.commands;

import net.thenamedev.legendapi.utils.Cooldown;
import net.thenamedev.legendapi.utils.Rank;
import net.thenamedev.legendarena.extras.menu.ParticleMenu;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.UUID;

/**
 * Particle selector command... of magic. (this seems to be a recuring theme of me saying "[x] of [y]" in javadocs)
 *
 * @author ThePixelDev
 */
public class Particle implements CommandExecutor {

    private HashMap<UUID, Cooldown> cooldown = new HashMap<>();

    public boolean onCommand(CommandSender sender, Command command, String s, String[] args) {
        if(!Rank.isRanked(sender, Rank.MEMBERPLUS)) {
            sender.sendMessage(Rank.noPermissions(Rank.MEMBERPLUS));
            return true;
        }
        if(cooldown.containsKey(((Player) sender).getUniqueId()) && !cooldown.get(((Player) sender).getUniqueId()).done()) {
            sender.sendMessage(cooldown.get(((Player) sender).getUniqueId()).getTimeRemaining());
            return true;
        }
        ParticleMenu.show((Player) sender);
        //2 second cooldown
        cooldown.put(((Player) sender).getUniqueId(), new Cooldown(2));
        return true;
    }

}

File: src/main/java/legendarena/commands/Particle.kt
package legendarena.commands

import legendapi.utils.Cooldown
import legendapi.utils.Rank
import legendapi.utils.RankUtils
import legendarena.hub.menu.ParticleMenu
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*

class Particle : CommandExecutor {

    private val cooldown = HashMap<UUID, Cooldown>()

    override fun onCommand(sender: CommandSender, command: Command, s: String, args: Array<String>): Boolean {
        if(!Rank.MEMBERPLUS.isRanked(sender)) {
            sender.sendMessage(RankUtils.noPermissions(Rank.MEMBERPLUS))
            return true
        }
        if(cooldown.containsKey((sender as Player).getUniqueId()) && !cooldown.get(sender.getUniqueId()).done()) {
            sender.sendMessage(cooldown.get(sender.getUniqueId()).getTimeRemaining())
            return true
        }
        ParticleMenu().show(sender)
        //1 second cooldown
        cooldown.put(sender.getUniqueId(), Cooldown(1.0))
        return true
    }

}